### PR TITLE
[refactor](decimal) change type name Decimal128 to Decimal128V2, Decimal128I to Decimal128V3 to avoid confusion

### PR DESCRIPTION
**Pull request by jacktengg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jacktengg** [comment]
run buildall

File: be/src/vec/core/field.cpp [hunk 1]
```diff
@@ -169,19 +169,19 @@ bool dec_less_or_equal(T x, T y, UInt32 x_scale, UInt32 y_scale) {
...
 DECLARE_DECIMAL_COMPARISON(Decimal64)
-DECLARE_DECIMAL_COMPARISON(Decimal128)
+DECLARE_DECIMAL_COMPARISON(Decimal128V2)
 DECLARE_DECIMAL_COMPARISON(Decimal256)
 
 template <>
-bool decimal_equal(Decimal128I x, Decimal128I y, UInt32 xs, UInt32 ys) {
-    return dec_equal(Decimal128(x.value), Decimal128(y.value), xs, ys);
+bool decimal_equal(Decimal128V3 x, Decimal128V3 y, UInt32 xs, UInt32 ys) {
+    return dec_equal(Decimal128V2(x.value), Decimal128V2(y.value), xs, ys);
```
Comment (yiguolei) [inline be/src/vec/core/field.cpp:177]: These code is very suspicious?

File: be/src/vec/data_types/data_type_factory.cpp [hunk 2]
```diff
@@ -311,13 +311,13 @@ DataTypePtr DataTypeFactory::create_data_type(const TypeIndex& type_index, bool
     case TypeIndex::Decimal64:
         nested = std::make_shared<DataTypeDecimal<Decimal64>>(BeConsts::MAX_DECIMAL64_PRECISION, 0);
         break;
-    case TypeIndex::Decimal128:
-        nested = std::make_shared<DataTypeDecimal<Decimal128>>(BeConsts::MAX_DECIMAL128_PRECISION,
-                                                               0);
+    case TypeIndex::Decimal128V2:
+        nested = std::make_shared<DataTypeDecimal<Decimal128V2>>(BeConsts::MAX_DECIMAL128_PRECISION,
+                                                                 0);
```
Comment (yiguolei) [inline be/src/vec/data_types/data_type_factory.cpp:316]: BeConsts::MAX_DECIMAL128_PRECISION maybe should rename this variable.
And also I think line314 is wrong , should use MAX_DECIMALV2_PRECISION?

**Comment by jacktengg** [comment]
run buildall

**Comment by yiguolei** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**